packages/e2e-tests/test/e2e.spec.ts

```diff
@@ -262,6 +262,26 @@ describe('e2e', function () {
       }
       expect(buffer).to.include('"i": 99999');
     });
+    it('handles custom prompt() function in conjunction with line-by-line input well', async function () {
+      // https://jira.mongodb.org/browse/MONGOSH-1617
+      shell = TestShell.start({
+        args: [
+          '--nodb',
+          '--shell',
+          '--eval',
+          'prompt = () => {sleep(1);return "x>"}',
+        ],
+      });
+      // The number of newlines here matters
+      shell.writeInput(
+        'sleep(100);print([1,2,3,4,5,6,7,8,9,10].reduce(\n(a,b) => { return a*b; }, 1))\n\n\n\n',
```

Comment:
probably silly question, but why is this sleeping longer if we want to make sure the prompt is evaluated first? also why does the number of newlines matter? 😅

Reply:
oh I get it now, thank you for the explanation!